includes/class-monk-links.php

```diff
@@ -0,0 +1,523 @@
+<?php
+/**
+ * The engine that changes links to use the apprpriate language
+ *
+ * @link       https://github.com/brenoalvs/monk
+ * @since      0.2.0
+ *
+ * @package    Monk
+ * @subpackage Monk/Includes
+ * @author     Alexandre Correa <alexandrecorreadeoliveira@gmail.com>
+ */
+
+/**
+ * This class holds all functions related to links transformation
+ * the permalinks are filtered to return content related to
+ * the language defined by the user.
+ *
+ * @package    Monk
+ * @subpackage Monk/Links
+ */
+class Monk_Links {
+
+	/**
+	 * The plugin ID.
+	 *
+	 * @since    0.2.0
+	 * @access   private
+	 * @var      string    $monk    The ID of this plugin.
+	 */
+	private $monk;
+
+	/**
+	 * The version of this plugin.
+	 *
+	 * @since    0.2.0
+	 * @access   private
+	 * @var      string    $version    The current version of this plugin.
+	 */
+	private $version;
+
+	/**
+	 * Initializes the class and set its properties.
+	 *
+	 * @since    0.2.0
+	 * @param    string $monk       The name and ID of the plugin.
+	 * @param    string $version    The plugin version.
+	 */
+	public function __construct( $monk, $version ) {
+		$structure = get_option( 'permalink_structure', false );
```

Comment:
What happens when `$structure` is `false`?
Why not use `monk_using_permalinks()` method here?